storage/innobase/handler/ha_innodb.cc

```diff
@@ -5633,8 +5633,21 @@ check_index_consistency(const TABLE* table, const dict_table_t* ib_table)
 			ret = false;
 			goto func_exit;
 		}
-	}
 
+		if (index->is_unique()) {
+			uint32_t idx_num = index->get_num();
```

Comment:
Do we really need a new member function for this?
```c++
if (index->is_unique()) {
	ulint i;
	while ((index = UT_LIST_GET_PREV(indexes, index))) i++;
	if (last_unique > i) {
		m_int_table_flags
			|= HA_DUPLICATE_KEY_NOT_IN_ORDER;
	}
	last_unique = i;
}
```
I’d use the same type for both; in your version, we are dealing with mismatching types `ulint` and `uint32_t`.